### PR TITLE
Improve error recovery in parser around explicit interface specifier in operators.

### PR DESCRIPTION
**Pull request by AlekseyTs**
Also adding a bunch of parsing tests for scenarios without modifiers.

### CODE REVIEW

File: src/Compilers/CSharp/Test/Syntax/Parsing/MemberDeclarationParsingTests.cs [hunk 1]
```diff
@@ -2409,5 +2357,1623 @@ public void OperatorDeclaration_ExplicitImplementation_20()
...
+                        }
+                    }
+                    N(SyntaxKind.SemicolonToken);
+                }
+                EOF();
+            }
+        }
+
+        [Fact]
+        public void OperatorDeclaration_ExplicitImplementation_23()
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Syntax/Parsing/MemberDeclarationParsingTests.cs:2456]: >OperatorDeclaration_ExplicitImplementation_23 [](http://example.com/codeflow?start=20&length=45)

Info: Starting from here the tests are just a clone of tests above with modifiers removed. CodeFlow shows really bad diff. There aren't that many changes to the tests above, I believe only two new tests were added and four modified there. Perhaps comparing sections by copying them locally will be more convenient.

**Review by 333fred — APPROVED** [review]
Thanks, the errors look much more sane now. LGTM (commit 1)

**Comment by AlekseyTs** [comment]
@RikkiGibson, @dotnet/roslyn-compiler For the second review.

**Review by RikkiGibson — APPROVED** [review]

File: src/Compilers/CSharp/Portable/Parser/LanguageParser.cs [hunk 2]
```diff
@@ -2286,6 +2286,7 @@ private MemberDeclarationSyntax ParseMemberDeclarationOrStatementCore(SyntaxKind
 
                 try
                 {
+                    // Try as a regualr statement rather than a member declaration, if appropriate.
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Portable/Parser/LanguageParser.cs:2289]: nit: typo on 'regular'

**Pull request merged**